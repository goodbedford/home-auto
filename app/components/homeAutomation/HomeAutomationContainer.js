import React from "react";
import {Route, Link, NavLink, Redirect} from "react-router-dom";
import HomeAutomation from "./HomeAutomation";
import HeaderContainer from "../header/HeaderContainer";
import HomeSection from "../homeSection/HomeSection";
import PanelItem from "../panelItem/PanelItem.js";
import PanelNav from "../panelNav/PanelNav.js";
import PanelDetails from "../panelDetails/PanelDetails.js";
import PanelControls from "../panelControls/PanelControls.js";
import PanelBlock from "../panelBlock/PanelBlock.js";
import Loading from "../loading/Loading.js";

import httpHelper from "../../utils/httpHelper.js";

const propTypes =  {
};
const defaultProps = {
};

class HomeAutomationContainer extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      isLoading: true,
      homes: []
    };
  }
  componentDidMount () {
    httpHelper.getHomes()
      .then(homes => {
        // console.log("homes", JSON.stringify(homes,null,2));
        this.setState({
          isLoading:false,
          homes: homes
        });
      })
      .catch(error => {
        console.warn("Error with getHomes", error);
      });
  }
  render() {
    if (this.state.isLoading) {
      return (
        <HomeAutomation>
          <HeaderContainer />
          <HomeSection>
            <Loading />
          </HomeSection>
        </HomeAutomation>
      );
    }
    return (
      <HomeAutomation>
        <HeaderContainer />
        <HomeSection>
          <PanelNav>
            <PanelControls header="Homes" match={this.props.match}  />
            <PanelItem items={this.state.homes} match={this.props.match} />;
          </PanelNav>
          <PanelDetails>
            <Route exact path="/homes" component={() => {
              return (
                <PanelBlock msg="Select a Home" />
              );
            }}/>
          </PanelDetails>
        </HomeSection>
      </HomeAutomation>
    );
  }
}
HomeAutomationContainer.propTypes = propTypes;
HomeAutomationContainer.defaultProps = defaultProps;

export default HomeAutomationContainer;
