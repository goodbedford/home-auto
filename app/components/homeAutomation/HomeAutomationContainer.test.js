/* eslint-disable no-undef */

import React from "react";
import {shallow} from "enzyme";
import toJson from "enzyme-to-json";
import HomeAutomationContainer from "./HomeAutomationContainer";

describe("<HomeAutomationContainer />", () => {
  it("HomeAutomationContainer snapshot", () => {
    const wrapper = shallow(<HomeAutomationContainer />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it("should have isLoading prop", () => {
    const wrapper = shallow(<HomeAutomationContainer />);
    // console.log(wrapperl)
    expect(wrapper.state("isLoading")).toEqual(true);
  });
  it("should have homes prop", () => {
    const wrapper = shallow(<HomeAutomationContainer />);
    const homes = wrapper.state("homes");
    expect(homes).toBeDefined();
    expect(Array.isArray(homes)).toEqual(true);
  });
});
