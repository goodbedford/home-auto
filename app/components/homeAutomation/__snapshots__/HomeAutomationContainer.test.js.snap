// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomeAutomationContainer /> HomeAutomationContainer snapshot 1`] = `
<HomeAutomation>
  <HeaderContainer />
  <HomeSection>
    <Loading
      loadingMsg="loading"
      styles="loading loading--block-md"
    />
  </HomeSection>
</HomeAutomation>
`;
